master/contrib/hgbuildbot.py

```diff
@@ -13,138 +13,228 @@
+#
+#     [hooks]
+#     changegroup.buildbot = python:.hg/hgbuiltbot.py:hook
+#
+#     [hgbuildbot]
+#     venv = /home/buildbot/.virtualenvs/builtbot/lib/python2.7/site-packages
+#     master = localhost:9987
+#     passwd = aA8(-adf8j3-_3uX
+#
+#
+# Available parmeters
+# -------------------
+#
+# venv
+#   The hook needs the Python package "buildbot".  You can optionally point to
+#   virtualenv if it is not installed globally:
+#
+#   Optional; default: None
+#
+#   Example:
+#
+#       venv = /path/to/venv/lib/pythonX.Y/site-packages
+#
+# master
+#   Host and port of the Buildmaster(s) to notify.
+#   Can be a single entry or a comma-separated list.
+#
+#   Mandatory.
+#
+#   Examples:
+#
+#       master = localhost:9989
+#       master = bm1.example.org:9989,bm2.example.org:9989
+#
+# user
+#   User for connecting to the Buildmaster.
+#
+#   Optional; default: change
+#
+# passwd
+#   Password for connecting to the Buildmaster.
+#
+#   Optional; default: changepw
+#
+# branchtype
+#   The branchmodel you use: "inrepo" for named branches (managed by
+#   "hg branch") or "dirname" for directory based branches (the last component
+#   of the repository's directory will then be used as branch name).
+#
+#   Optional; default: inrepo
+#
+# branch
+#   Explicitly specify a branchname instead of using the repo's basename when
+#   using "branchtype = dirname".
+#
+#   Optional.
+#
+# baseurl
+#   Prefix for the repository URL sent to the Buildmaster.  See below for
+#   details.
+#
+#   Optional.  The hook will also check the [web] section for this parameter.
+#
+# strip
+#   Strip as many slashes from the repo dir before appending it to baseurl.
+#   See below for details.
+#
+#   Optional; default: 0; The hook will also check the [notify] section for
+#   this parameter.
+#
+# category
+#   Category to assign to all change sets.
+#
+#   Optional.
+#
+# project
+#   Project that the repo belongs to.
+#
+#   Optional.
+#
+# codebase
+#   Codebase name for the repo.
+#
+#   Optional.
+#
+#
+# Repository URLs
+# ---------------
+#
+# The hook sends a repository URL to the Buildmasters.  It can be used by
+# schedulers (e.g., for filtering) and is also used in the webview to create
+# a link to the corresponding changeset.
+#
+# By default, the absolute repository path (e.g., "/home/hg/repos/myrepo") will
+# be used.  The webview will in this case simply append the path to its own
+# hostname in order to create a link to that change (e.g.,
+# "http://localhost:8020/home/hg/repos/myrepo").
+#
+# You can alternatively strip some of the repo path's components and prepend
+# a custom base URL instead.  For example, if you want to create an URL like
+# "https://code.company.com/myrepo", you must specify the following parameters:
+#
+#     baseurl = https://code.company.com/
+#     strip = 4
+#
+# This would strip everything until (and including) the 4th "/" in the repo's
+# path leaving only "myrepo" left.  This would then be append to the base URL.
```

Comment:
Thank you for providing an extensive documentation.

I believe one little thing is missing: what needs to be configured on the Buildbot side.  I believe a reference to [PBChangeSource](http://docs.buildbot.net/latest/manual/cfg-changesources.html#pbchangesource) would be appropriate.